test/tests/functional/pbs_soft_walltime.py

```diff
@@ -0,0 +1,634 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+def cvt_duration(duration):
+    """
+    convert string form of a duration (HH:MM:SS) into seconds
+    """
+    h = 0
+    m = 0
+    sp = duration.split(':')
+    if len(sp) == 3:
+        h = int(sp[0])
+        m = int(sp[1])
+        s = int(sp[2])
+    elif len(sp) == 2:
+        m = int(sp[0])
+        s = int(sp[1])
+    else:
+        s = int(sp[0])
+
+    return h * 3600 + m * 60 + s
+
+
+class TestSoftWalltime(TestFunctional):
+    """
+    Test that the soft_walltime resource is being used properly and
+    being extended properly when exceeded
+    """
+
+    def compare_estiamtes(self, running_job, jobs):
```

Comment:
Why this function name contains 'estamtes' than 'estimates'? This looks wierd

Reply:
It was a typo.  I fixed it.